pkg/scheduler/apis/config/validation/validation_test.go

```diff
@@ -336,28 +375,43 @@ func TestValidatePolicy(t *testing.T) {
 					{Name: "customPriority2", Weight: 2, Argument: &config.PriorityArgument{ServiceAntiAffinity: &config.ServiceAntiAffinity{}}},
 				},
 			},
-			expected: errors.New("ServiceAntiAffinity  priority \"customPriority2\" has a different weight with \"customPriority1\""),
+			expected: &field.Error{
+				Type:     field.ErrorTypeInvalid,
+				Field:    "priorities[1].argument.serviceAntiAffinity.weight",
+				BadValue: int64(2),
+				Detail:   "priority has a different weight with \"customPriority1\"",
+			},
 		},
 		{
 			name: "invalid hardPodAffinitySymmetricWeight, above the range",
 			policy: config.Policy{
 				HardPodAffinitySymmetricWeight: 101,
 			},
-			expected: errors.New("hardPodAffinitySymmetricWeight: Invalid value: 101: not in valid range [0-100]"),
+			expected: &field.Error{
+				Type:     field.ErrorTypeInvalid,
+				Field:    "hardPodAffinitySymmetricWeight",
+				BadValue: int32(101),
+				Detail:   "not in valid range [0-100]",
+			},
 		},
 		{
 			name: "invalid hardPodAffinitySymmetricWeight, below the range",
 			policy: config.Policy{
 				HardPodAffinitySymmetricWeight: -1,
 			},
-			expected: errors.New("hardPodAffinitySymmetricWeight: Invalid value: -1: not in valid range [0-100]"),
+			expected: &field.Error{
+				Type:     field.ErrorTypeInvalid,
+				Field:    "hardPodAffinitySymmetricWeight",
+				BadValue: int32(-1),
+				Detail:   "not in valid range [0-100]",
+			},
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			actual := ValidatePolicy(test.policy)
-			if fmt.Sprint(test.expected) != fmt.Sprint(actual) {
+			if !errors.Is(actual, test.expected) {
```

Comment:
I'm not sure errors.Is is a good match for a parameterized error like this. for unit test purposes, the existing checks didn't seem bad to me

Reply:
Note that `Aggregate` implements `Is`, but it's not useful if the actual errors don't.